### DISCUSSION TITLE
NextJS 14.1 now supports window.history.replaceState

### DISCUSSION THREAD

**Discussion by shawngustaw** [47ng/nuqs#466] [Ideas] [answered]
Just wondering if there's any plans to update `nuqs` to use these new api's?

Release info is here: https://nextjs.org/blog/next-14-1#windowhistorypushstate-and-windowhistoryreplacestate

**Comment by franky47** [comment, marked as answer]
Ah, `nuqs` has been using the Window History API since v1.8.0, much before Next.js did actually support it. That was the only way to do shallow updates (and it still is), which is the default.

The only difference now is that the app router is aware of shallow URL changes, and won't reset them accidentally (as it was the case in `next@14.0.2` with the prefetch bug, see #388).

It's been a wild ride, and a couple of Next.js versions had issues while they were implementing proper support for the History API, but I'm glad it's all converged now. I wrote about all of this [on my blog](https://francoisbest.com/posts/2023/storing-react-state-in-the-url-with-nextjs).
